builder/dockerfile/internals.go

```diff
@@ -489,9 +489,13 @@ func hostConfigFromOptions(options *types.ImageBuildOptions, isWCOW bool) *conta
 	// For WCOW, the default of 20GB hard-coded in the platform
 	// is too small for builder scenarios where many users are
 	// using RUN statements to install large amounts of data.
-	// Use 127GB as that's the default size of a VHD in Hyper-V.
-	if isWCOW {
-		hc.StorageOpt = make(map[string]string)
+	// If no custom size is set, use 127GB for containers that
+	// are started during build, as that's the default size of
+	// a VHD in Hyper-V.
+	if isWCOW && hc.StorageOpt["size"] == "" {
+		if hc.StorageOpt == nil {
```

Comment:
`hc.StorageOpt` can't be nil here?

Reply:
Had to double-check, but yes, it can be; https://play.golang.org/p/bD7H35U2wdZ

```go
package main

import (
	"fmt"
)

type HostConfig struct {
	StorageOpt map[string]string `json:",omitempty"` // Storage driver options per container.
}

func main() {
	hc := &HostConfig{}

	if hc.StorageOpt["size"] == "" {
		if hc.StorageOpt == nil {
			fmt.Println("StorageOpt is nil")
		}

		hc.StorageOpt["size"] = "127GB"
	}
}
```

Running that will panic;

```
StorageOpt is nil
panic: assignment to entry in nil map

goroutine 1 [running]:
main.main()
	/tmp/sandbox898184674/prog.go:19 +0xa8
```